backend/routes/referralRouters.ts

```diff
@@ -15,6 +16,53 @@ referralRouter.get(
   async (req: Request, res: Response, next: NextFunction) => {}
 );
 
+referralRouter.get(
+  "/user",
+  async (req: Request, res:Response, next: NextFunction) => {
+    try{
+      const userId = req.query.userId ? (req.query.userId as string) : "";
+      if(req.user?.id!==userId){ //we should only get referrals if the user logged is the user passed into get request
+        throw new StatusCodedError("unauthorized request: userId does not match", 401);
+      }
+      const referrals = await referralControllers.getReferralsFromUserId(userId);
```

Comment:
The idea makes sense to me, but the implementation does not. 
Instead of trying to get a userId field from the frontend, why don't the backend just use the authenticated userId, which is already stored in req.user.id?